lib/spack/spack/package_prefs.py

```diff
@@ -252,5 +253,76 @@ def is_spec_buildable(spec):
     return allpkgs[spec.name]['buildable']
 
 
+def get_package_dir_permissions(spec):
+    """Return the permissions configured for the spec.
+
+    Include the GID bit if group permissions are on. This makes the group
+    attribute sticky for the directory."""
+    perms = get_package_permissions(spec)
+    if perms & stat.S_IRWXG:
+        perms |= stat.S_ISGID
+    return perms
+
+
+def get_package_permissions(spec):
+    """Return the permissions configured for the spec"""
+    allpkgs = get_packages_config()
+
+    # Get readable permissions level
+    if spec.name in allpkgs and 'readable' in allpkgs[spec.name]:
+        readable = allpkgs[spec.name]['readable']
+    elif 'all' in allpkgs and 'readable' in allpkgs['all']:
+        readable = allpkgs['all']['readable']
+    else:
+        readable = 'world'
+    if readable not in ('user', 'group', 'world'):
```

Comment:
Don't hard-code things like this.  Possible levels like `('user', 'group', 'world')` should be defined as module variables, similar to how dependency types are defined in `dependency.py`.

Also, this should probably just be checked with schema validation rather than explicitly in the code.